CMakeLists.txt

```diff
@@ -1,61 +1,179 @@
-        endif()
-        set(Boost_USE_STATIC_RUNTIME ON)
-    endif()
+project(${PACKAGE_NAME} LANGUAGES CXX DESCRIPTION "The QuantLib C++ Library")
+
+# Path for package-local cmake modules
+set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake" ${CMAKE_MODULE_PATH})
+
+# Installation directories
+set(QL_INSTALL_BINDIR "bin" CACHE STRING "Installation directory for executables")
+set(QL_INSTALL_LIBDIR "lib" CACHE STRING "Installation directory for libraries")
+set(QL_INSTALL_INCLUDEDIR "include" CACHE STRING "Installation directory for headers")
+set(QL_INSTALL_EXAMPLESDIR "lib/QuantLib/examples" CACHE STRING
+    "Installation directory for examples")
+
+# Options
+option(QL_BUILD_BENCHMARK "Build benchmark" ON)
+option(QL_BUILD_EXAMPLES "Build examples" ON)
+option(QL_BUILD_TEST_SUITE "Build test suite" ON)
+option(QL_ENABLE_OPENMP "Detect and use OpenMP" OFF)
+option(QL_ENABLE_PARALLEL_UNIT_TEST_RUNNER "Enable the parallel unit test runner" OFF)
+option(QL_ENABLE_SESSIONS "Singletons return different instances for different sessions" OFF)
+option(QL_ENABLE_SINGLETON_THREAD_SAFE_INIT "Enable thread-safe singleton initialization" OFF)
+option(QL_ENABLE_THREAD_SAFE_OBSERVER_PATTERN "Enable the thread-safe observer pattern" OFF)
+option(QL_ENABLE_TRACING "Tracing messages should be allowed" OFF)
+option(QL_ERROR_FUNCTIONS "Error messages should include current function information" OFF)
+option(QL_ERROR_LINES "Error messages should include file and line information" OFF)
+option(QL_EXTRA_SAFETY_CHECKS "Extra safety checks should be performed" OFF)
+option(QL_HIGH_RESOLUTION_DATE "Enable date resolution down to microseconds" OFF)
+option(QL_INSTALL_BENCHMARK "Install benchmark" ON)
+option(QL_INSTALL_EXAMPLES "Install examples" ON)
+option(QL_INSTALL_TEST_SUITE "Install test suite" ON)
+option(QL_TAGGED_LAYOUT "Library names use layout tags" ${MSVC})
+option(QL_USE_DISPOSABLE "Use the Disposable class template. Not needed for C++11" OFF)
+option(QL_USE_INDEXED_COUPON "Use indexed coupons instead of par coupons" OFF)
+option(QL_USE_STD_CLASSES "Enable all QL_USE_STD_ options" OFF)
+option(QL_USE_STD_SHARED_PTR "Use standard smart pointers instead of Boost ones" OFF)
+option(QL_USE_STD_UNIQUE_PTR "Use std::unique_ptr instead of std::auto_ptr" ON)
+option(QL_USE_STD_FUNCTION "Use std::function and std::bind instead of Boost ones" OFF)
+option(QL_USE_STD_TUPLE "Use std::tuple instead of boost::tuple" OFF)
+
+# Convenience option to activate all STD options
+if (QL_USE_STD_CLASSES)
+    set(QL_USE_STD_SHARED_PTR ON)
+    set(QL_USE_STD_UNIQUE_PTR ON)
+    set(QL_USE_STD_FUNCTION ON)
+    set(QL_USE_STD_TUPLE ON)
+endif()
+
+# Sessions requires thread-safe initialization
+if (QL_ENABLE_SESSIONS)
+    set(QL_ENABLE_SINGLETON_THREAD_SAFE_INIT ON)
+endif()
+
+# Shared or static libs
+if (NOT DEFINED BUILD_SHARED_LIBS)
+    set(BUILD_SHARED_LIBS OFF)
```

Comment:
This could be 
```
    set(BUILD_SHARED_LIBS ${UNIX})
```
to keep the existing behavior

Reply:
I'm fine with that --  will change.